CompileError.py

```diff
@@ -1,3 +1,38 @@
+from traceback import print_exception
+from pathlib import Path
+from contextvars import ContextVar
+
+cur_file_var = ContextVar("cur_file_var",default=None)
+
+class Pos:
```

Comment:
Maybe use @dataclasses.dataclass from the standard library here (https://docs.python.org/3/library/dataclasses.html)